src/tests/integration/test_generic_db_operations.py

```diff
@@ -770,27 +753,140 @@ def test_sql_query(ip_with_dynamic_db, cell, request, test_table_name_dict):
     assert out.error_in_exec is None
 
 
-@pytest.mark.parametrize("ip_with_dynamic_db", ALL_DATABASES)
-def test_sql_query_cte_suggestion(ip_with_dynamic_db, request):
+@pytest.mark.parametrize(
+    "cell",
+    [
+        "%%sql\nSELECT * FROM subset",
+        "%%sql --with subset\nSELECT * FROM subset",
+    ],
+    ids=[
+        "cte-inferred",
+        "cte-explicit",
+    ],
+)
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_postgreSQL",
+        "ip_with_mySQL",
+        "ip_with_mariaDB",
+        "ip_with_SQLite",
+        "ip_with_duckDB_native",
+        "ip_with_duckDB",
+        pytest.param(
+            "ip_with_MSSQL",
+            marks=pytest.mark.xfail(
+                reason="We need to close any pending results for this to work"
+            ),
+        ),
+        "ip_with_Snowflake",
+        "ip_with_oracle",
+    ],
+)
+def test_sql_query_cte(ip_with_dynamic_db, request, test_table_name_dict, cell):
     ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+
     ip_with_dynamic_db.run_cell(
-        """%%sql --save first_cte --no-execute
-SELECT 1 AS column1, 2 AS column2
-"""
-    )
-    ip_with_dynamic_db.run_cell(
-        """
-    %%sql --save second_cte --no-execute
-SELECT
-  sum(column1),
-  sum(column2) FILTER (column2 = 2)
-FROM first_cte
-"""
+        "%%sql --save subset --no-execute \n"
+        f"SELECT * FROM {test_table_name_dict['numbers']}"
     )
+
+    out = ip_with_dynamic_db.run_cell(cell)
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_postgreSQL",
+        "ip_with_mySQL",
+        "ip_with_mariaDB",
+        "ip_with_SQLite",
+        pytest.param(
+            "ip_with_duckDB_native",
+            marks=pytest.mark.xfail(reason="Not yet implemented"),
+        ),
+        "ip_with_duckDB",
+        "ip_with_Snowflake",
+        pytest.param(
+            "ip_with_MSSQL", marks=pytest.mark.xfail(reason="Not yet implemented")
+        ),
+        pytest.param(
+            "ip_with_oracle", marks=pytest.mark.xfail(reason="Not yet implemented")
+        ),
+    ],
+)
+def test_sql_error_suggests_using_cte(ip_with_dynamic_db, request):
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+
     out = ip_with_dynamic_db.run_cell(
         """
     %%sql
-SELECT * FROM second_cte"""
+S"""
     )
     assert isinstance(out.error_in_exec, UsageError)
+    assert out.error_in_exec.error_type == "RuntimeError"
     assert CTE_MSG in str(out.error_in_exec)
```

Comment:
Not related to original issue, but is possible you can fix extra [space char in the CTE_MSG](https://github.com/ploomber/jupysql/blob/fabf7cfa1a6b4f26332d0a40a50e0ebcb6e06d6e/src/sql/error_message.py#L4)?

From:
```
 "For more details please refer : "
```
To:
```
 "For more details please refer: "
```